### PR TITLE
#41: [Optimizer] Create `update` endpoint

### PR DESCRIPTION
**Pull request by MitchellGray100**
Implements: #41

### CODE REVIEW

File: microservices/optimizer/app.py [hunk 1]
```diff
@@ -38,8 +38,40 @@
...
+            if name is None or time is None:
+                return (
+                    jsonify({"error": "missing test class name or execution time"}),
+                    400,
+                )
+
+            # Update database (Fixed in SPEED-41)
+
+            app.logger.info(
+                f"Updated {name} in {url} on branch {branch} with time {time}"
```
[the remark this answers is not in the document]
Comment (MitchellGray100) [reply microservices/optimizer/app.py]: what do u want me to do about this

File: microservices/optimizer/app.py [hunk 2]
```diff
@@ -38,8 +38,40 @@ def update_times() -> tuple[Response, int]:
...
+            if name is None or time is None:
+                return (
+                    jsonify({"error": "missing test class name or execution time"}),
+                    400,
+                )
+
+            # Update database (Fixed in SPEED-41)
+
+            app.logger.info(
+                "Updated {name} in {url} on branch {branch} with time {time}"
```
Comment (jakejack13) [inline microservices/optimizer/app.py:65]: Switch to f-string
```suggestion
                f"Updated {name} in {url} on branch {branch} with time {time}"
```

Comment (MitchellGray100) [reply microservices/optimizer/app.py:65]: I had that beforehand but mypy said nonono

**Comment by MitchellGray100** [comment]
@jakejack13 Could you please run make format and commit the changes please? Thank you

**Comment by jakejack13** [comment]
@MitchellGray100 please merge in the new changes from main and I'll review

**Review by jakejack13 — APPROVED** [review]
LGTM

**Pull request merged**